articles/quickstart/spa/vuejs/_includes/_centralized_login.md

```diff
@@ -2,475 +2,441 @@
 
 ```js
 // src/router.js
 
 //.. other imports
 import Profile from "./views/Profile.vue";
 
-const routes = [
-  {
-    path: '/callback',
-    name: 'callback',
-    component: Callback
-  },
-
-  // NEW - add the route to the /profile component
-  {
-    path: "/profile",
-    name: "profile",
-    component: Profile
-  }
-];
-
-// Unchanged code
 const router = new VueRouter({
   mode: 'history',
-  routes
+  routes: [
+    // .. other routes and pages ..
+
+    // NEW - add the route to the /profile component
+    {
+      path: "/profile",
+      name: "profile",
+      component: Profile
+    }
+  ]
 });
 
 export default router;
 ```
 
-Then add the `/profile` route to your navigation bar by inserting a new `<li>` element into the navigation bar structure:
+Next, modify the navigation bar in `App.vue` to include a route to the profile page, making sure that it is only available when the user is authenticated:
 
 ```html
-// src/App.vue
-
-//... other navigation code
-
-<li>
-  <router-link to="/">Home</router-link>
-</li>
-<li v-if="!isAuthenticated">
-  <a href="#" @click.prevent="login">Login</a>
-</li>
-
-<!-- new link to /profile - only show if authenticated -->
-<li v-if="isAuthenticated">
-  <router-link to="/profile">Profile</router-link>
-</li>
-<!-- /profile -->
-
-<li v-if="isAuthenticated">
-  <a href="#" @click.prevent="logout">Log out</a>
-</li>
+<template>
+  <div id="app">
+    <div id="nav">
+      <router-link to="/">Home</router-link>|
+      <router-link to="/about">About</router-link>|
+
+      <!-- NEW - add a route to the profile page -->
+      <router-link v-if="$auth.isAuthenticated" to="/profile">Profile</router-link>
+    </div>
+    <router-view />
+  </div>
+</template>
 ```
 
-### Securing the profile route
+:::panel Checkpoint
+Now when you run the application, you should be able to log in and see the link to the profile page in the navigation bar. When you navigate to the profile page, the user's information will be visible on the screen.
+:::
+
+## Secure the Profile Page
 
 Even though the `/profile` route is only shown if the user is authenticated, the user could still manually type the URL into the browser and access the page if they have not logged in — although there will be nothing to see.
 
 A catch-all rule can be added to the router so that access is only permitted if the user is logged in. If they are not logged in, they will be prompted to log in before being redirected to the location they tried to access in the first place.
 
-Open `router.js` and add a rule that exhibits this behavior:
+Create a new file in the `src/auth` folder called `authGuard.js`, and use the `getInstance` method of the authentication service to implement a function that will prevent a route from being accessed by an unauthenticated user:
 
 ```js
-// src/router.js
+import { getInstance } from "./authService";
```

Comment:
Getting an error with this:

```
Failed to compile.

./src/auth/authGuard.js
Module not found: Error: Can't resolve 'authService' in '/Users/josh-cunningham/Sites/quickstarts/spa-qs-vue/src/auth'
```

Reply:
Ah, the import was wrong. I've fixed it now

Reply:
The change:

```js
import { getInstance } from "./auth";
```

... still gave me a compile error. I changed to:

```js
import { getInstance } from "./index";
```

... and that worked.

Reply:
Ah of course, because you're in a file in the same folder. Thanks for the correction!